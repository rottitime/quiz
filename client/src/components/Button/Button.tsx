import React from "react"
import { Link } from "react-router-dom";

type Props = {
    to: string,
    children: React.ReactNode
}

const Button: React.FC<Props> = ({ to, children }) => {
    return <Link to={to} className="waves-effect waves-light btn-large">
        {children}
    </Link>
}


export default Button