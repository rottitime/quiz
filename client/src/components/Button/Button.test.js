import React from 'react'
import Button from './Button'
import { render, cleanup } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup)

it("renders button correctly", () => {
    const { getByText } = render(<Router>
        <Button to="/somewhere">Click me</Button>
    </Router>)
    expect(getByText('Click me')).toBeInTheDocument()
    // expect(container.firstChild).toMatchInlineSnapshot(`<h1>Hello, World!</h1>`)
})

it("matches snapshot", () => {
    const tree = renderer.create(<Router>
        <Button to="/somewhere">Click me</Button>
    </Router>).toJSON()

    expect(tree).toMatchSnapshot()
})