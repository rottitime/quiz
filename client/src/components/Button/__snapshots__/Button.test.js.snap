// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  className="waves-effect waves-light btn-large"
  href="/somewhere"
  onClick={[Function]}
>
  Click me
</a>
`;
